### DISCUSSION TITLE
Disable/hide collection  based on role

### DISCUSSION THREAD

**Discussion by Eshaftri** [directus/directus#15902] [Archive]
I'm wondering if it is possible to disable/hide collection from the context menu based on the role of the user accesses.

**Comment by azrikahar** [comment]
Can you share which specific context menu you are referring to? If a role does not have read permission to a collection (via Roles & Permissions), the collection should be hidden for that role.

**Reply by NeoKactus** [reply]
He could be talking about the collection navigation list on the content module page. I have found it a bit awkward to hide collections for specific roles.

Example Collections:
Movies - Visible and has a relationship to Genres
Genres - Hidden from collection list

The issue comes from if we want our role to still see the Genres field through the Movies collection. They require read access to the Genres collection, but that will cause it to show up in the navigation list.

Our current solution is to use a collection folder and set it to hidden. That way, the permissions are happy and we can hide whatever we want. The main issue with this is that it hides the collection for every role and is an annoyance for admins. 

It would be nice if we can set 'read' permissions for collection folders per role.

**Reply by connorwinston** [reply]
If that is in fact what he is referring too, I have seen someone share an extension that matches that behavior...if I find it floating around I will add it here.

Also, here is another discussion referencing that...https://github.com/directus/directus/discussions/11233

**Reply by Pachat** [reply]
> If that is in fact what he is referring too, I have seen someone share an extension that matches that behavior...if I find it floating around I will add it here.
@ConnorSimply Hi, Did you find the extension? I very much need such a feature.
